dist/angular-date-picker.js

```diff
@@ -22,92 +22,121 @@ angular.module( "vokal.datePicker", [] )
     {
         "use strict";
 
-        var defaultFormat = "M/d/yyyy";
+        var defaultFormat = "M/D/YYYY";
 
-        function validateDate( date )
+        function validateDate( dateOrString )
         {
-            return !!date && angular.isFunction( date.getTime ) && !isNaN( date.getTime() );
-        }
-        function convertToDate( str )
-        {
-            return validateDate( str ) ? str : new Date( str );
+            return !!dateOrString && moment( new Date( dateOrString ) ).isValid();
         }
 
         return {
             restrict: "A",
-            scope: {},
+            scope: {
+                model: "=ngModel",
+                timezone: "="
+            },
             require: "ngModel",
             link: function ( scope, element, attrs, ngModelController )
             {
-                var localDate = new Date( new Date().toDateString() );
-                function filterOutput( date )
+                if( scope.timezone && !moment.tz )
+                {
+                    console.warn( "Trying to use moment-timezone without including the script." );
+                }
+                var localMoment = scope.timezone ? moment.tz() : moment();
+
+                function filterForModel()
                 {
                     return attrs.pickerType === "string" ?
-                        $filter( "date" )( date, attrs.datePicker || defaultFormat ) : date;
+                        filterForRender( localMoment ) :
+                        localMoment.toDate();
+                }
+                function filterForRender( dateMoment )
+                {
+                    return dateMoment.format( attrs.datePicker || defaultFormat );
+                }
+
+                function setLocalDate( month, day, year )
+                {
+                    if( moment.isDate( month ) )
```

Comment:
I would just split this function into one for setting by date and one for setting by month, day, year. The overloading is adding complexity.